Sources/llbuild2fx/SpawnProcess.swift

```diff
@@ -0,0 +1,345 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+import Foundation
+import TSCBasic
+import TSCUtility
+import TSFCAS
+import TSFCASFileTree
+import TSFFutures
+
+extension Foundation.Process {
+    private enum ProcessTerminationError: Error {
+        case nonZeroExit(Int32)
+        case signaled(Int32)
+        case unknown(Int32)
+    }
+
+    private struct ProcessKiller: LLBCancelProtocol {
+        private let runningProcess: Foundation.Process
+        init(runningProcess: Foundation.Process) {
+            self.runningProcess = runningProcess
+        }
+
+        func cancel(reason: String?) {
+            let pid = runningProcess.processIdentifier
+            kill(pid, SIGKILL)
+        }
+    }
+
+    fileprivate func runLater(_ ctx: Context) -> LLBCancellableFuture<Void> {
```

Comment:
`runCancellable`?    i.e. it is spawning it immediately, but wrapping it in a cancellable future.